torch/csrc/utils/schema_info.cpp

```diff
@@ -191,12 +206,42 @@ std::vector<c10::FunctionSchema> SchemaInfo::getNonDeterministicOps() {
   return nondeterministic_ops;
 }
 
+void SchemaInfo::initSchemaInfo() {
+  std::unordered_set<at::Symbol> seen;
+  auto init_schema_arguments =
+      [this, &seen](
+          const std::vector<c10::Argument>& arguments_list,
+          c10::SchemaArgType type) {
+        seen.clear();
+        for (size_t i = 0; i < arguments_list.size(); i++) {
+          if (arguments_list[i].alias_info()) {
+            if (arguments_list[i].alias_info()->isWildcardAfter()) {
+              wildcard_set_.insert({type, i});
+            } else {
+              for (const auto& set :
+                   arguments_list[i].alias_info()->afterSets()) {
+                TORCH_INTERNAL_ASSERT(
```

Comment:
is this accurate for returns?

I agree for inputs, it doesn't really make sense to say that two inputs are aliasing (except for in some obscure after-set aliasing where the op would swap data pointers...)

But for outputs, it seems reasonable that you might have two outputs that both alias the same input... for example you might imagine `split_in_half(Tensor(a) x) -> (Tensor(a), Tensor(a))`

either way, can you add some comments on why we have this assertion?

Reply:
It's mainly because the check would not be accurate in cases where two outputs both alias the same output and all three are not wildcards. At least from the way I'm currently representing the alias graph, accounting for this case would require going from linear / constant time operations to full dfs searches (up to n^2 time). Furthermore, no ops currently have this aliasing pattern so the assert is to ensure that the class will be accurate in the future (ie: no one runs an op with the above aliasing pattern expecting correct results). 

I'll add a better worded version of the above as a comment in the code.